File: 3.1/3.1/Clock.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _3._1
{
    public class Clock
    {
        private Counter _hours;
        private Counter _minutes;
        private Counter _seconds;

        public Clock()
        {
            _hours = new Counter("Hours");
            _minutes = new Counter("Minutes");
            _seconds = new Counter("Seconds");
        }

        public void Tick()
        {
            _seconds.Increment();
            if (_seconds.Ticks == 60)
            {
                _seconds.Reset();
                _minutes.Increment();
                if (_minutes.Ticks == 60)
                {
                    _minutes.Reset();
                    _hours.Increment();
                    if (_hours.Ticks == 24)
                    {
                        _hours.Reset();
                        
                    }
                }
            }
        }

        public void Reset()
        {
            _hours.Reset();
            _minutes.Reset();
            _seconds.Reset();
        }

        public string Display()
        {
            Console.WriteLine($"{ _hours.Ticks:D2}: {_minutes.Ticks:D2}: {_seconds.Ticks:D2}");
            return $"{ _hours.Ticks:D2}: {_minutes.Ticks:D2}: {_seconds.Ticks:D2}";
        }
    }
}

File: CLock in a new language/Clock.py
class Counter:
    def __init__(self, name):
        self._count = 0
        self._name = name

    def increment(self):
        self._count += 1

    def reset(self):
        self._count = 0

    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, value):
        self._name = value

    @property
    def ticks(self):
        return self._count


class Clock:
    def __init__(self):
        self._hours = Counter("Hours")
        self._minutes = Counter("Minutes")
        self._seconds = Counter("Seconds")

    def tick(self):
        self._seconds.increment()
        if self._seconds.ticks == 60:
            self._seconds.reset()
            self._minutes.increment()
            if self._minutes.ticks == 60:
                self._minutes.reset()
                self._hours.increment()
                if self._hours.ticks == 24:
                    self._hours.reset()

    def reset(self):
        self._hours.reset()
        self._minutes.reset()
        self._seconds.reset()

    def display(self):
        display_str = f"{self._hours.ticks:02}:{self._minutes.ticks:02}:{self._seconds.ticks:02}"
        print(display_str)
        return display_str


def main():
    clock = Clock()

    for i in range(90000):
        clock.tick()
        if i % 86400 == 0:
            clock.reset()
        clock.display()


if __name__ == "__main__":
    main()
